models.py

```diff
@@ -1,4 +1,115 @@
+import bcrypt
+from flask import redirect, render_template, url_for
+import os
+from app import app
 from flask_sqlalchemy import SQLAlchemy
-from flask_login import UserMixin
+from flask_login import UserMixin, login_user, LoginManager, login_required, logout_user
+from wtforms import StringField, SubmitField, PasswordField
+from flask_wtf import FlaskForm
+from wtforms.validators import InputRequired, Length, ValidationError
+from flask_bcrypt import Bcrypt
 
-db = SQLAlchemy()
+
+db = SQLAlchemy(app)
+bcrypt = Bcrypt(app)
+
+
+
+login_manager = LoginManager()
+login_manager.init_app(app)
+login_manager.login_view ="login"
+
+
+@login_manager.user_loader
+def load_user(user_id):
+    return Signup.query.get(int(user_id))
+
+class Signup(db.Model, UserMixin):
```

Comment:
Since this is basically the table representing users, this should probably be renamed to "Users" for clarity's sake. Any parts of other blocks of code here that uses "Signup" should also be changed to "Users" as well (ex: the queries made for "Signup").